server/src/main/java/org/opensearch/action/search/FetchSearchPhase.java

```diff
@@ -240,6 +240,7 @@ private void executeFetch(
                     public void innerOnResponse(FetchSearchResult result) {
                         try {
                             progressListener.notifyFetchResult(shardIndex);
+                            context.setPhaseResourceUsages();
```

Comment:
Why do we need to handle this separately during FetchSearchPhase execution, but not other implementations of SearchPhase?

Reply:
because other phases extend AbstractSearchAsyncAction, so we can simply add setPhaseResourceUsages for all of them in AbstractSearchAsyncAction. However `FetchSearchPhase` extends `SearchPhase` only,